### ISSUE TITLE
HTTP clients don't generate correctly when using a CommonType

### ISSUE THREAD

**Issue by paprikati** [goadesign/goa#3099]
When we generate our HTTP client from a design that uses a `CommonType`, the http `types.go` files still looks in the specific service package, instead of common.

e.g.
```
// NewBuildApplicableFieldsResponseBody builds the HTTP response body from the
// result of the "BuildApplicableFields" endpoint of the "Incidents" service.
func NewBuildApplicableFieldsResponseBody(res *incidents.IncidentApplicableFields) *BuildApplicableFieldsResponseBody {
	body := &BuildApplicableFieldsResponseBody{}
	return body
}
```

Should be 
```
// NewBuildApplicableFieldsResponseBody builds the HTTP response body from the
// result of the "BuildApplicableFields" endpoint of the "Incidents" service.
func NewBuildApplicableFieldsResponseBody(res *common.IncidentApplicableFields) *BuildApplicableFieldsResponseBody {
	body := &BuildApplicableFieldsResponseBody{}
	return body
}
```

**Comment by raphael** [comment]
Thanks for the report, I coincidently just merged a few fixes in this area (although mostly with gRPC so might be unrelated). Could you please try with the latest v3 and if that still does not work please provide a simple design that reproduces the issue. To try with the latest you can use the following command:
```bash
go get goa.design/goa/v3@6be5979
```
Thank you!

**Comment by terryvogelsang** [comment]
Hi there,

First thanks a lot for the great work!

I am encountering the same behavior but in the HTTP server generation. Please see below for a simple reproducible example.

Hope this helps. Thanks!

## Setup
Tested with Goa version `v3.8.3`
### Layout
```
.
├── api
│   └── design
│       └── design.go
├── go.mod
└── go.sum
```

### Content of `design.go`
```go
package design

import (
	. "goa.design/goa/v3/dsl"
)

// Main API declaration
var _ = API("myapp", func() {
	Version("1.0")
	Title("MyApp REST API")
	Server("myapp", func() {
		Services("account")
		Host("localhost", func() {
			Description("default host")
			URI("http://localhost:8005/")
		})
	})
})

// CustomError is the custom API error response
var CustomError = Type("CustomError", func() {
	Meta("struct:pkg:path", "common")
	ErrorName("code", String, "Code of the error", func() {
		Example("not_found")
	})
	Required("code", "message")
})

// Account is the custom type for accounts Response
var Account = Type("Account", func() {
	Description("Account describes an account in the application.")
	Attribute("id", String, "ID is the unique resource identifier.", func() {
		Example("d1877801-099a-475e-9d27-3a8472c46543")
	})
	Attribute("account_name", String, "Name of the account.", func() {
		Example("johndoe")
	})
	Required("id", "account_name")
})

// Account Service declaration with methods and API specification file
var _ = Service("account", func() {
	Error("not_found", CustomError)

	HTTP(func() {
		Response("not_found", StatusNotFound)
	})
	Description("The account service allows access to account management features")

	Method("read_account_by_id", func() {
		Payload(func() {
			Field(1, "account_id", String, "Account ID", func() {
				Example("2018b387-95ac-456e-b3a9-c5e899251ca8")
			})
			Required("account_id")
		})
		Result(Account)
		HTTP(func() {
			GET("/api/v1/accounts/{account_id}")
			Response(StatusOK)
		})
	})
})


```

## Problem
When generating files with `goa gen minimal-repro/api/design -o api`, `api/gen/http/account/server/types.go` contains the following: 

```go
// Code generated by goa v3.8.3, DO NOT EDIT.
//
// account HTTP server types
//
// Command:
// $ goa gen minimal-repro/api/design -o api

package server

import (
	account "minimal-repro/api/gen/account"
)

// ReadAccountByIDResponseBody is the type of the "account" service
// "read_account_by_id" endpoint HTTP response body.
type ReadAccountByIDResponseBody struct {
	// ID is the unique resource identifier.
	ID string `form:"id" json:"id" xml:"id"`
	// Name of the account.
	AccountName string `form:"account_name" json:"account_name" xml:"account_name"`
}

// ReadAccountByIDNotFoundResponseBody is the type of the "account" service
// "read_account_by_id" endpoint HTTP response body for the "not_found" error.
type ReadAccountByIDNotFoundResponseBody struct {
	// Code of the error
	Code string `form:"code" json:"code" xml:"code"`
}

// NewReadAccountByIDResponseBody builds the HTTP response body from the result
// of the "read_account_by_id" endpoint of the "account" service.
func NewReadAccountByIDResponseBody(res *account.Account) *ReadAccountByIDResponseBody {
	body := &ReadAccountByIDResponseBody{
		ID:          res.ID,
		AccountName: res.AccountName,
	}
	return body
}

// NewReadAccountByIDNotFoundResponseBody builds the HTTP response body from
// the result of the "read_account_by_id" endpoint of the "account" service.
func NewReadAccountByIDNotFoundResponseBody(res *account.CustomError) *ReadAccountByIDNotFoundResponseBody {
	body := &ReadAccountByIDNotFoundResponseBody{
		Code: res.Code,
	}
	return body
}

// NewReadAccountByIDPayload builds a account service read_account_by_id
// endpoint payload.
func NewReadAccountByIDPayload(accountID string) *account.ReadAccountByIDPayload {
	v := &account.ReadAccountByIDPayload{}
	v.AccountID = accountID

	return v
}
```

We can see that it indeed still looks in the specific service package (`account.CustomError`), instead of common (`common.CustomError`). To allow the expected behavior, it should also import the package `minimal-repro/api/gen/common`

**Comment by raphael** [comment]
Thank you for the great report and repro. This is fixed by #3127.
